patterns/1-initial/innersource-hackathon.md

```diff
@@ -0,0 +1,80 @@
+
+* From middle management: perception of no direct benefit to the team apart from team upskilling and individual growth; eventually resulting in no motivation by the middle management to the engineers in trying out InnerSource ways of working.
+* From engineers: investing time to try new ways of working when there is already time and resource constraint for regular work deliverables; it is also challenging to balance both and show outcomes in both; this curbs motivation from the engineers as there is little safe space to experiment.
+* From senior leadership: not prioritizing investment in InnerSource ways of working such as it being part of long term goals or OKRs.
+
+For those new to InnerSource or any technology/methodology, there is a need for safe space to try and experiment without definite outcomes in the initial stages. These factors and forces hinder such an experiment ground.
+
+## Solution
+
+Organize a company wide hackathon focused on InnerSource. An InnerSource hackathon provides a safe space for engineers to try and contribute to InnerSource projects without any presumption and prejudice. It could be a 1 or 2 day event.
+
+It can preferrable be organized by InnerSource Program Office (ISPO), if it exists in the organization or by the Open Source Program Office (OSPO), if OSPO also drives InnerSource within the company. It can also be organized by other common service organizations too, if need be. Basically a central team or a group of individuals, who believe in InnerSource, can organize the hackathon.
+
+All engineers in the organization can participate in the hackathon. The participants can be new to InnerSource, or InnerSource practitioners already. They can participate individually or as a team. Participating as a team also provides a safe environment, for example, those are new to InnerSource can team up InnerSource practitioners.
+
+There could be one or more categories to participate as follows:
+
+* Start a new InnerSource project: It could be either starting a new project from scratch as InnerSource or making a existing project InnerSource ready.
+* Contribute to the existing InnerSource project: InnerSource project owners can list down the features in GitHub issues as a pre-requisite before the hackathon and the participants can contribute to it during the hackathon.
+* Participate as an InnerSource mentor: Participants can also mentor new InnerSource practitioners and provide knowledge transfer or coaching during the event.
+
+The event could be held virtually or physically in different locations in the organization based on the cost and budget involved. A virtual event will have best ROI but it could depend on each organization.
+
+The winners and participants should be recognized and acknowledged in a company wide forum at the end of the hackathon. This is important as it keeps motivating them and more engineers to adopt and practice InnerSource going forward.
```

Comment:
```suggestion
The winners and participants should be recognized and acknowledged in a company-wide forum at the end of the hackathon. This is important as it keeps motivating them and more engineers to adopt and practice InnerSource going forward.
```

Reply:
I have organized a number of hackathons in the past, and for company-internal hackathons the part of "winning" was always controversial. Some people loved the friendly competition, others thought that selecting a winner was harmful.

Personally I like the friendly competition :)

For an InnerSource hackathon in particular I am curious:
How did you select the winners? Which criteria did you apply?

Reply:
Yes, it is tough to set the right judging criteria :) 

We had two categories in the hackathon - 'Start a new InnerSource project' and 'Contribute to existing InnerSource projects'. In the contributing category, as a pre-requisite, the innersource maintainers listed down the GitHub issues for which they were open for contributions. The participants can contribute to one of these issues or make their own contribution.

Judging criteria for 'Start a new InnerSource project': 
- How well the InnerSource project aligns with our internal guidelines for an InnerSource project (InnerSource readiness)? 
- What is the value proposition? How does it benefit the many in the company? 
- How welcoming and open the repository is for new contributors? 

Judging criteria for 'Contribute to an existing InnerSource project': 
- Participating team should contribute to an InnerSource project that they don't own. 
- Value-add the contribution brings to the project. 
- How well it aligns with the project?
- How well the contributor has aligned with the contributing guidelines in the project? 

The judging criteria was shared with the judging panel as well as the participants prior to the event. The winners were selected based on these criteria. We also declared winners per category as it would not be fair to compare outcome of different categories.